mirrord/intproxy/src/proxies/incoming/interceptor.rs

```diff
@@ -156,41 +179,59 @@ impl HttpConnection {
-                    tracing::warn!(
-                        "Sending request to local application failed with: {e:?}. \
+            Err(InterceptorError::Hyper(e)) if e.is_closed() => {
+                tracing::warn!(
+                    "Sending request to local application failed with: {e:?}. \
                         Seems like the local application closed the connection too early, so \
                         creating a new connection and trying again."
-                    );
-                    tracing::trace!("The request to be retried: {request:?}.");
+                );
+                tracing::trace!("The request to be retried: {request:?}.");
 
-                    Err(InterceptorError::ConnectionClosedTooSoon(request))
-                }
+                Err(InterceptorError::ConnectionClosedTooSoon(request))
+            }
 
-                Err(InterceptorError::Hyper(e)) if e.is_parse() => {
-                    tracing::warn!("Could not parse HTTP response to filtered HTTP request, got error: {e:?}.");
-                    let body_message = format!("mirrord: could not parse HTTP response from local application - {e:?}");
-                    Ok(HttpResponseFallback::response_from_request(
+            Err(InterceptorError::Hyper(e)) if e.is_parse() => {
+                tracing::warn!(
+                    "Could not parse HTTP response to filtered HTTP request, got error: {e:?}."
+                );
+                let body_message = format!(
+                    "mirrord: could not parse HTTP response from local application - {e:?}"
+                );
+                Ok((
+                    HttpResponseFallback::response_from_request(
                         request,
                         StatusCode::BAD_GATEWAY,
                         &body_message,
-                    ))
-                }
+                    ),
+                    None,
+                ))
+            }
 
-                Err(err) => {
-                    tracing::warn!("Request to local application failed with: {err:?}.");
-                    let body_message = format!("mirrord tried to forward the request to the local application and got {err:?}");
-                    Ok(HttpResponseFallback::response_from_request(
+            Err(err) => {
+                tracing::warn!("Request to local application failed with: {err:?}.");
+                let body_message = format!(
+                    "mirrord tried to forward the request to the local application and got {err:?}"
+                );
+                Ok((
+                    HttpResponseFallback::response_from_request(
                         request,
                         StatusCode::BAD_GATEWAY,
                         &body_message,
-                    ))
-                }
+                    ),
+                    None,
+                ))
+            }
+
+            Ok(mut res) if matches!(request, HttpRequestFallback::Framed(_)) => {
```

Comment:
Would consider moving the logic of the match statement to 
```rust
fn handle_hyper_response<F, B>(&self, request: &HttpRequestFallback, res: hyper::Response<hyper::body::Incoming>, response_map: F) -> (HttpResponseFallback, Option<OnUpgrade>)
  where F: Fn(B) -> HttpResponseFallback {
    ...
     HttpResponse::<B>::from_hyper_response(
      res,
      self.peer.port(),
      request.connection_id(),
      request.request_id()
     )
      .await
      .map(response_map)
      .unwrap_or_else(|e| {
      ...
  }
```

Because it looks very similar to the next match

Reply:
Managed to merge both branches ^^